src/reanimated2/layoutReanimation/web/animationsManager.ts

```diff
@@ -0,0 +1,485 @@
+}
+
+/**
+ * Creates transition of given type, appends it to stylesheet and returns keyframe name.
+ *
+ * @param transitionType Type of transition (e.g. LINEAR).
+ * @param transitionData Object containing data for transforms (translateX, scaleX,...).
+ * @returns Keyframe name that represents transition.
+ */
+export function TransitionGenerator(
+  transitionType: TransitionType,
+  transitionData: TransitionData
+) {
+  const keyframe = generateRandomKeyframeName();
+  let transition;
+
+  switch (transitionType) {
+    case TransitionType.LINEAR:
+      transition = LinearTransition(keyframe, transitionData);
+      break;
+    case TransitionType.SEQUENCED:
+      transition = SequencedTransition(keyframe, transitionData);
+      break;
+    case TransitionType.FADING:
+      transition = FadingTransition(keyframe, transitionData);
+      break;
+  }
+
+  const styleTag = document.getElementById(
+    WEB_ANIMATIONS_ID
+  ) as HTMLStyleElement;
+
+  if (styleTag.sheet) {
+    styleTag.sheet.insertRule(transition);
+  } else {
+    console.error(
+      '[Reanimated] Failed to insert layout animation into CSS stylesheet'
+    );
+  }
+
+  return keyframe;
+}
+
+export function tryActivateLayoutTransition<
+  ComponentProps extends Record<string, unknown>
+>(
+  props: Readonly<AnimatedComponentProps<ComponentProps>>,
+  element: HTMLElement,
+  snapshot: DOMRect
+) {
+  const rect = element.getBoundingClientRect();
+
+  if (areDOMRectsEqual(rect, snapshot)) {
+    return;
+  }
+
+  const transitionData: TransitionData = {
+    translateX: snapshot.x - rect.x,
+    translateY: snapshot.y - rect.y,
+    scaleX: snapshot.width / rect.width,
+    scaleY: snapshot.height / rect.height,
+    reversed: false, // This field is used only in `SequencedTransition`, so by default it will be false
+  };
+
+  handleWebAnimation(
+    props,
+    element,
+    LayoutAnimationType.LAYOUT,
+    transitionData
+  );
+}
+
+export function handleWebAnimation<
+  ComponentProps extends Record<string, unknown>
+>(
+  props: Readonly<AnimatedComponentProps<ComponentProps>>,
+  element: HTMLElement,
+  animationType: LayoutAnimationType,
+  transitionData?: TransitionData
+) {
+  const config =
+    animationType === LayoutAnimationType.ENTERING
+      ? props.entering
+      : animationType === LayoutAnimationType.EXITING
+      ? props.exiting
+      : animationType === LayoutAnimationType.LAYOUT
+      ? props.layout
+      : null;
+
+  if (!config) {
+    return;
+  }
+
+  const isLayoutTransition = animationType === LayoutAnimationType.LAYOUT;
+
+  const initialAnimationName =
+    typeof config === 'function' ? config.name : config.constructor.name;
+
+  // This prevents crashes if we try to set animations that are not defined.
+  // We don't care about layout transitions since they're created dynamically
+  if (!(initialAnimationName in Animations) && !isLayoutTransition) {
+    if (props.entering) {
+      element.style.visibility = 'initial';
```

Comment:
In that specific case `element` will have visibility `initial`.

I've added this code yesterday because I've found out that after I set visibility to `hidden` (when component has `entering` prop) and for some reason animation fails (for example it wasn't implemented - at this point it can be either `custom` or `keyframe` animation), the visibility was never reset. 

Now when I think about it, anytime when someone renders `hidden` component it will become visible after animation finishes. I'll check that out later.

Reply:
After talking with @piaskowyk we decided that we will keep it as it is. We don't see much sense in setting `entering` animation and then hiding that component.